packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -1329,6 +1330,28 @@ export default class NewVm extends BaseComponent {
 
   // DISKS -----------------------------------------------------------------------
 
+  _getSrsNotOnSameHost = createSelector(
+    () => this.state.state.existingDisks,
+    () => this.state.state.VDIs,
+    () => this.props.srs,
+    (existingDisks, vdis, srs) => {
+      let container
+      const predicate = srId =>
+        container !== undefined
+          ? container !== srs[srId].$container
+          : ((container = get(() => srs[srId].$container)), false)
+
+      return (
+        vdis.some(({ SR: srId }) => srId !== null && predicate(srId)) &&
+        (isEmpty(existingDisks) ||
```

Comment:
The complexity of this algorithme is `O(N2)`. Maybe my proposition above can reduce it.